### ISSUE TITLE
[Regression] Emoticon picker is being opened by groupchat title change in detached window mode

### ISSUE THREAD

**Issue by zetok** [qTox/qTox#2208]
Description: Emoticon picker is being opened every time groupchat title changes, after it is being opened by user even once.

OS: Hardened Gentoo Linux x86_64
i3 wm
qTox: 7dd66213cbef6d6a6d7d6b2ecc3f89f9d8d8b7b9

Reproducible: Always

Steps to reproduce:
1. Set `Open chats in separate window` in settings.
2. In groupchat, click on emoticon picker.
3. Change groupchat title, emoticon picker is being opened by it.

Observed Behavior:
Emoticon picker is being opened by groupchat title change.

Expected Behavior:
Emoticon picker should not be opened by groupchat title change.

Additional info:
https://d.maxfile.ro/yxpkasoptm.webm

Reproducible for me on i3wm on Gentoo, and for @ovalseven8:

> Can reproduce it. Tested with latest Lubuntu version.

First reported in https://github.com/tux3/qTox/pull/1938#issuecomment-117553543

**Comment by sudden6** [comment]
Not reproducible on 76c4cf7d390ece3d0ec57b771fb9aaa21f7337f7 with plasmashell 5.7.3 on openSUSE Tumbleweed. @zetok  can you still reproduce?

**Comment by zetok** [comment]
Still reproducible on 4a5f89bb98fc8657514e35c5022713358a453d49

**Comment by sudden6** [comment]
then I need more info, on how to reproduce.

**Comment by zetok** [comment]
Have you actually changed the groupchat title? Note that when you are alone in a "groupchat" qTox doesn't change its title..

**Comment by sudden6** [comment]
Yes, I did it exactly like in your steps. I used two clients on one pc and the different environment.

**Comment by zetok** [comment]
Perhaps me using i3wm matters

**Comment by sudden6** [comment]
Maybe you can try on a different DE?

**Comment by zetok** [comment]
Can reproduce on XFCE @ OpenSUSE Tubleweed VM on qTox 1.5.0

**Comment by zetok** [comment]
on qTox 76c4cf7d390ece3d0ec57b771fb9aaa21f7337f7 webm: 
[topic-emoji-picker.zip](https://github.com/qTox/qTox/files/453899/topic-emoji-picker.zip)

**Comment by sudden6** [comment]
I still can't reproduce on XFCE @ OpenSUSE Tumbleweed VM
[emoticon-picker.zip](https://github.com/qTox/qTox/files/453960/emoticon-picker.zip)

**Comment by sudden6** [comment]
I could reproduce now, but I have no idea what exactly the issue is or how to fix it. So I'll write down what I found out so far.
- The desktop Environment seems to be irrelevant, I could finally reproduce on Plasma 5 and XFCE.
- it is not reproducible with a simple groupchat between two qTox clients.
- the only groupchat where it was reproducible for me was qgroup@toxme.io, there were 6 people online including the bots at the time of testing.
- title was `qTox groupchat | note that groupchat history is not kept | synced with #qtox @ freenode | there are public log` but it doesn't seem to matter, because if the title was copied to another groupchat it didn't happen anymore.
- one active nick contained a `\0` char at the time of testing, no idea if it's relevant, because the issue continued to  happen even when the groupchat split and the tested client was the only one in the groupchat.
- [Enter] has to be pressed after editing the group title for the problem to occur

Problem:
Somehow the keypress [Enter] event for the groupchat title widget also triggers https://github.com/qTox/qTox/blob/76c4cf7d390ece3d0ec57b771fb9aaa21f7337f7/src/widget/form/genericchatform.cpp#L370

The issue seems to be present since around 7dd66213cbef6d6a6d7d6b2ecc3f89f9d8d8b7b9.

**Comment by Diadlo** [comment]
~Seems, after creation `ChatFormHeader`, call button pressed, but not smiles.~
In a AV group chat or in a friend dialog, call button pressed. In group chat without AV smiles button pressed. It's the reason why you can't reproduce in manually created group.

After debugging, I can suggest that it's result of behaviour like focus stealing: when you press Enter `CroppingLabel` hides `QLineEdit` and shows `QLabel`. Focus was on `QLineEdit` and probably changed on the next element of the form after => button. And for some reason it activate it too.

It maybe Qt bug or not
